engine/common/mod_bmodel.c

```diff
@@ -207,6 +210,151 @@ static mlumpinfo_t		extlumps[EXTRA_LUMPS] =
 { LUMP_SHADOWMAP, 0, MAX_MAP_LIGHTING / 3, sizeof( byte ), -1, "shadowmap", USE_EXTRAHEADER, (const void **)&srcmodel.shadowdata, &srcmodel.shadowdatasize },
 };
 
+/*
+===============================================================================
+
+			Static helper functions
+
+===============================================================================
+*/
+
+static mip_t* Mod_GetMipTexForTexture( dbspmodel_t* bmod, int textureIndex )
+{
+	if( textureIndex < 0 ||
+	     textureIndex >= bmod->textures->nummiptex ||
+	     bmod->textures->dataofs[textureIndex] == -1 )
+	{
+		return NULL;
+	}
+
+	return (mip_t*)((byte*)bmod->textures + bmod->textures->dataofs[textureIndex]);
+}
+
+// Returns index of WAD that texture was found in, or -1 if not found.
+static int Mod_FindTextureInWadList(
+	wadlist_t* list,
+	const char* textureName,
+	char* foundPathBuffer,
+	size_t foundPathBufferLength )
+{
+	char pathInWad[MAX_VA_STRING];
+	int wadIndex;
+
+	if( !list || !textureName || !(*textureName) )
+	{
+		return -1;
+	}
+
+	// check wads in reverse order
+	for( wadIndex = list->count - 1; wadIndex >= 0; --wadIndex )
+	{
+		Q_snprintf( pathInWad, sizeof(pathInWad), "%s.wad/%s.mip", list->wadnames[wadIndex], textureName );
+
+		if( FS_FileExists( pathInWad, false ) )
+		{
+			if( foundPathBuffer && foundPathBufferLength > 0 )
+			{
+				Q_strncpy( foundPathBuffer, pathInWad, foundPathBufferLength );
+			}
+
+			return wadIndex;
+		}
+	}
+
+	return -1;
+}
+
+static size_t Mod_CalculateMipTexSize( mip_t* mipTex, qboolean customPalette )
```

Comment:
You can use shorter variables, so expressions won't take multiple lines for nothing.

Also, the asterisk must be on the right side.